src/yunohost/certificate.py

```diff
@@ -731,13 +731,7 @@ def _dns_ip_match_public_ip(public_ip, domain):
 
 def _domain_is_accessible_through_HTTP(ip, domain):
     try:
-        # Check HTTP reachability
         requests.head("http://" + ip, headers={"Host": domain})
```

Comment:
no matter what `domain` value is, this requests will not return any exception ;)

Maybe we should request /yunohost/admin or /yunohost/api to be sure that domain & ip will match. See https://dev.yunohost.org/issues/638 for example